### ISSUE TITLE
About the clip_gradient parameter

### ISSUE THREAD

**Issue by hezw2016** [huangzehao/caffe-vdsr#27]
hi, huang
I made the clip_gradient = 0.1. when training, it always shows like this
"
I0111 17:21:27.510517  4417 sgd_solver.cpp:92] Gradient clipping: scaling down gradients (L2 norm 1.03171 > 0.1) by scale factor 0.0969264
"
from **the begining to the end**. (The numbers are different).
Check the original paper, the authors used the adjustable clip_gradient parameter. I think that your code is different from the original paper in the aspect of clip_gradient parameter. Will it cause some problem?
By the way, your code still gets a good performance.
Thanks for sharing your code with us.

**Comment by huangzehao** [comment]
Hi, first, the clip_gradient implementation in caffe is different from other framworks, such as torch or mxnet.
In caffe, clip_gradient is done at the norm of all gradients. But in torch or mxnet, it's done at the value of gardient in every layer. Read the codes of clip_gradient and you will know it. And I guess the clip_gradient implementation of author is similar as torch's or mxnet's implementation.  

Second, the author didn't offer the accurate value of theta in his paper. I choose 0.1 because the change of gradient is stable in this setting. You can try increasing the clip_gradient value when decreasing the learning rate. I did't do much experiments for turning parameters, but you can have a try.
Thanks.

**Comment by hezw2016** [comment]
Thx for your reply.
